### PR TITLE
Refactor code and remove Unirest library

### PR DESCRIPTION
**Pull request by Shairam**

### CODE REVIEW

File: components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java [hunk 1]
```diff
@@ -112,24 +106,17 @@ private JSONObject getClientCredentials() throws OIDCProviderException {
...
-                    log.debug("Error occured while closing input stream reader ", e);
-                }
-            }
         }
     }
 
+    /**
+     * This method will be used to construct request body for DCR registeration
+     *
+     * @param dcrEp        This will be the DCR URL.
```
Comment (nadundesilva) [inline components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java]: Don't add "." at the end of the param and return description. Also add a "." at the end of the method description.

File: components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java [hunk 2]
```diff
@@ -175,30 +174,99 @@ private JSONObject retrieveClientCredentials(String dcrEp, HttpClient client) th
...
         return true;
     }
+
+    /**
+     * This method will be used to construct and make the request for validating token .
+     *
+     * @param token This will be the access token.
+     * @return the response object is returned.
+     */
+    private HttpResponse makeRequestforTokenValidity(String token) throws OIDCProviderException {
```
Comment (nadundesilva) [inline components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java]: The method should be "makeRequestforTokenValidation"

File: components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java [hunk 3]
```diff
@@ -175,30 +174,99 @@ private JSONObject retrieveClientCredentials(String dcrEp, HttpClient client) th
...
-                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
-                    .basicAuth(CelleryConfig.getInstance().getIdpAdminUsername()
-                            , CelleryConfig.getInstance().getIdpAdminPassword()).body("token=" + token).asString();
-
-            JSONObject jsonResponse = new JSONObject(stringResponse.getBody());
-            if (stringResponse.getStatus() != 200) {
+            HttpResponse response = makeRequestforTokenValidity(token);
+            StringBuilder builder = responseStringBuilder(response);
+            JSONObject jsonObject = new JSONObject(builder.toString());
+            if (response.getStatusLine().getStatusCode() != 200) {
```
Comment (nadundesilva) [inline components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java]: This check should be statusCode >= 200 && statusCode < 400 with follow redirects enabled for the HttpClient

Comment (Shairam) [reply components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java]: The HttpClient by default does the follow redirection.

File: components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java [hunk 4]
```diff
@@ -175,30 +174,100 @@ private JSONObject retrieveClientCredentials(String dcrEp, HttpClient client) th
...
-                    .basicAuth(CelleryConfig.getInstance().getIdpAdminUsername()
-                            , CelleryConfig.getInstance().getIdpAdminPassword()).body("token=" + token).asString();
-
-            JSONObject jsonResponse = new JSONObject(stringResponse.getBody());
-            if (stringResponse.getStatus() != 200) {
+            HttpResponse response = makeRequestforTokenValidition(token);
+            StringBuilder builder = getResponseString(response);
+            JSONObject jsonObject = new JSONObject(builder.toString());
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode >= 200 && statusCode < 400) {
```
Comment (nadundesilva) [inline components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java]: Shouldn't this check change ? This check is for successful response. IINM it should be the error response according to the if block body

**Pull request merged**